fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -0,0 +1,244 @@
+                        Document document = new Document();
+                        byte[] primaryKey = savedRecord.getPrimaryKey().pack();
+                        BytesRef ref = new BytesRef(primaryKey);
+                        document.add(new StoredField(PRIMARY_KEY_FIELD_NAME, ref));
+                        document.add(new SortedDocValuesField(PRIMARY_KEY_SEARCH_NAME, ref));
+                        for (int i = 0; i < fieldNames.size(); i++) {
+                            document.add(new TextField(fieldNames.get(i), entry.getKey().getString(i), Field.Store.NO));
+                        }
+                        writer.addDocument(document);
+                    }
+                } catch (IOException ioe) {
+                    throw new RuntimeException(ioe);
+                }
+            });
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        return AsyncUtil.DONE;
+    }
+
+    @Nonnull
+    @Override
+    public RecordCursor<IndexEntry> scanUniquenessViolations(@Nonnull TupleRange range, @Nullable byte[] continuation, @Nonnull ScanProperties scanProperties) {
+        LOG.debug("scanUniquenessViolations");
+        return RecordCursor.empty();
+    }
+
+    @Nonnull
+    @Override
+    public RecordCursor<InvalidIndexEntry> validateEntries(@Nullable byte[] continuation, @Nullable ScanProperties scanProperties) {
+        LOG.debug("validateEntries");
+        return RecordCursor.empty();
+    }
+
+    @Override
+    public boolean canEvaluateRecordFunction(@Nonnull IndexRecordFunction<?> function) {
+        LOG.debug("canEvaluateRecordFunction() function={}", function);
+        return false;
+    }
+
+    @Nonnull
+    @Override
+    public <T, M extends Message> CompletableFuture<T> evaluateRecordFunction(@Nonnull EvaluationContext context,
+                                                                              @Nonnull IndexRecordFunction<T> function,
+                                                                              @Nonnull FDBRecord<M> record) {
+        LOG.warn("evaluateRecordFunction() function={}", function);
+        return unsupportedRecordFunction(function);
+    }
+
+    @Override
+    public boolean canEvaluateAggregateFunction(@Nonnull IndexAggregateFunction function) {
+        LOG.trace("canEvaluateAggregateFunction() function={}", function);
+        return false;
+    }
+
+    @Nonnull
+    @Override
+    public CompletableFuture<Tuple> evaluateAggregateFunction(@Nonnull IndexAggregateFunction function,
+                                                              @Nonnull TupleRange range,
+                                                              @Nonnull IsolationLevel isolationLevel) {
+        LOG.warn("evaluateAggregateFunction() function={}", function);
+        return unsupportedAggregateFunction(function);
+    }
+
+    @Override
+    public boolean isIdempotent() {
+        LOG.trace("isIdempotent()");
+        return false;
```

Comment:
Why not? Doesn't saving the same record twice overwrite the document?

Reply:
yes it does overwrite the document.